internals/servicelog/ringbuffer.go

```diff
@@ -158,6 +166,10 @@ func (rb *RingBuffer) Size() int {
 func (rb *RingBuffer) Positions() (start RingPos, end RingPos) {
 	rb.rwlock.RLock()
 	defer rb.rwlock.RUnlock()
+	return rb.positions()
+}
+
+func (rb *RingBuffer) positions() (start RingPos, end RingPos) {
```

Comment:
We could just get rid of `positions` entirely, as the only other place it's used is in `TailIterator`, where we use the read index and throw away the write index. Could just use `rb.readIndex` directly there.

Similar with `closed` -- could just get rid of that and use `rb.writeClosed` directly in the couple of places it's needed.